### DISCUSSION TITLE
Improve behaviour for Calendar component in range mode

### DISCUSSION THREAD

**Discussion by galimovkirill** [primefaces/community#2033] [PrimeReact] [answered]
Calendar component do not provide visual display for selected range of **month** or **years** with range mode.
For now, it works for days only.

![image](https://github.com/primefaces/community/assets/55671052/a77011a4-576f-4695-918b-15ffd6326cbe)

Furthermore for years, it displays only start of range as selected.

![image](https://github.com/primefaces/community/assets/55671052/89eaf683-db48-47b7-b25f-e4fe72ed8408)

I suggest render years and months with styles as days have.

![image](https://github.com/primefaces/community/assets/55671052/57a5653d-c411-4adc-aff0-cd7c1e8c4997)

**Comment by KirilCycle** [comment, marked as answer]
Hi, this issue has already been raised. it will be addressed in the future. You can track its progress here: [#6653](https://github.com/primefaces/primereact/issues/6653).
